File: php/testcode/benchmark_test_00191.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00191(BenchmarkRequest $req): BenchmarkResponse {
    $allowedDomains = ['api.example.com', 'cdn.example.com', 'images.example.com'];
    $url = $req->param('url');
    $host = parse_url($url, PHP_URL_HOST);
    if (!in_array($host, $allowedDomains, true)) {
        return BenchmarkResponse::badRequest("Domain not allowed");
    }
    $content = file_get_contents($url);
    return BenchmarkResponse::ok($content);
}

File: rust/testcode/benchmark_test_00191.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let user_url = req.param("avatar_url");

    let html = format!("<html><body><img src='{}' alt='avatar'></body></html>", user_url);

    super::shared::BenchmarkResponse::ok(&html)
}
